### PR TITLE
ncm-directoryservices ncm-fmonagent ncm-mcx: Deprecate with warnings

### PR DESCRIPTION
**Pull request by jrha**
Throw deprecation if the following components are initialised:

* ncm-directoryservices
* ncm-fmonagent
* ncm-mcx

This is the first step towards #1027.

### CODE REVIEW

File: ncm-directoryservices/src/main/pan/components/directoryservices/schema.pan [hunk 1]
```diff
@@ -24,5 +24,7 @@ type component_directoryservices = {
         "ldapv3" ? directoryservices_ldap_entry{}
 };
 
-bind "/software/components/directoryservices" = component_directoryservices;
-
+bind "/software/components/directoryservices" = component_directoryservices with {
```
Comment (stdweird) [inline ncm-directoryservices/src/main/pan/components/directoryservices/schema.pan]: any reason why this is done at the bind, not at the type itself? (not that i care 😄 )

Comment (jrha) [inline ncm-directoryservices/src/main/pan/components/directoryservices/schema.pan]: It just feels more appropriate at the point it is actually used.

**Comment by jrha** [comment]
Is Jenkins okay? It seems like it's taking forever to build PRs at the moment.

**Comment by stdweird** [comment]
ran out of inodes, we're cleaning up atm

**Comment by stdweird** [comment]
retest this please

**Comment by jrha** [comment]
:green_salad: !

**Comment by jrha** [comment]
@stdweird I'll let you merge this if you want to do the deprecation in this release cycle.

**Comment by stdweird** [comment]
@jrha i want to, but i'll let @ned21 decide, as this also deprecates syslogng

**Comment by jrha** [comment]
Discussed at March 2017 workshop... no longer deprecates syslogng as it is expected to live on for some time at RAL and MS.

**Comment by jrha** [comment]
@stdweird this should be good to merge now

**Pull request merged**